src/http_workers/worker.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Result;
+use std::sync::Arc;
+use tokio::sync::{mpsc, oneshot, Mutex};
+
+use super::Work;
+
+pub struct Worker<W> {
```

Comment:
Worker does not have to be `pub` at all. The user of the Pool will never see this structure. The public interface of the workers Pool is both the Pool and WorkerHandler